### PR TITLE
feat(projects.project): add disco proj update to full & claim voucher

### PR DESCRIPTION
**Pull request by brofik**
ref: TAPC-96

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-discovery` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-96 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/activate/activate.controller.js [hunk 1]
```diff
@@ -2,14 +2,33 @@ import illustration from './assets/activate-bg.png';
 
 export default class ProjectActivateController {
   /* @ngInject */
-  constructor($q, orderCart, pciProjectNew) {
+  constructor($q, orderCart, pciProjectNew, CucCloudMessage, $translate) {
     this.$q = $q;
     this.orderCart = orderCart;
     this.pciProjectNew = pciProjectNew;
+    this.CucCloudMessage = CucCloudMessage;
+    this.$translate = $translate;
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/activate/activate.controller.js:10]: You don't use $translate, you can remove it

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js [hunk 2]
```diff
@@ -9,11 +11,90 @@ export default /* @ngInject */ ($stateProvider) => {
...
         $transition$.params().projectId || null,
 
-      goToProjectDashboard: /* @ngInject */ (goToState, projectId) => () => {
-        return goToState({
-          state: 'pci.projects.project',
-          params: { projectId },
-        });
+      activateProject: /* @ngInject */ (
+        serviceId,
+        projectId,
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/activate/activate.routing.js:16]: You can remove this

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js [hunk 3]
```diff
@@ -9,11 +11,90 @@ export default /* @ngInject */ ($stateProvider) => {
...
+        const voucherPayload = {
+          code: DISCOVERY_PROMOTION_VOUCHER,
+        };
+
+        return claimDiscoveryVoucher(voucherPayload)
+          .then(() => projectService.activateDiscoveryProject(serviceId))
+          .then(() => goToProjectDashboard(discoveryPromotionVoucherAmount))
+          .catch((err) => {
+            return displayErrorMessage(err.data.message);
+          });
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/activate/activate.routing.js:32]:
```suggestion
          .catch((err) => 
            displayErrorMessage(err.data.message);
          );
```

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js [hunk 4]
```diff
@@ -9,11 +11,90 @@ export default /* @ngInject */ ($stateProvider) => {
...
+      },
+
+      claimDiscoveryVoucher: /* @ngInject */ (projectService, projectId) => (
+        data,
+      ) => {
+        return projectService.claimVoucher(projectId, data);
+      },
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/activate/activate.routing.js:39]:
```suggestion
      ) => 
        projectService.claimVoucher(projectId, data);
      ,
```

File: packages/manager/modules/pci/src/projects/project/project.component.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import template from './project.html';
 export default {
   bindings: {
     projectId: '<',
+    serviceId: '<',
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/project.component.js:7]: We already have projectId. Can't we use same ?

Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/project.component.js:7]: in this context service and project are two diffrent entities.

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js (hunk 4 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/activate.routing.js]: done

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js (hunk 3 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/activate.routing.js]: done

File: packages/manager/modules/pci/src/projects/project/activate/activate.routing.js (hunk 2 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/activate.routing.js]: done

**Review by qpavy — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -1,16 +1,22 @@
...
+  "pci_projects_project_activate_banner_promotion_description_2": "Vous serez débité uniquement si vous poursuivez l'utilisation de nos produits au delà de cette somme ou cette date. De quoi démarrer sereinement.",
+  "pci_projects_project_activate_banner_no_promotion_description": "Nos services sont consommables à la demande. Vous pouvez facilement démarrer un service pour le tester quelques heures et l’éteindre dès que vous n’en avez plus besoin.",
+  "pci_projects_project_activate_banner_section_1_title": "Facturation granulaire",
+  "pci_projects_project_activate_banner_section_1_description": "La facturation des ressources se fait en fonction de votre utilisation. Si vous utilisez un produit une heure, vous ne serez facturé que de cette heure là.",
+  "pci_projects_project_activate_banner_section_2_title": "Facturation simple",
+  "pci_projects_project_activate_banner_section_2_description": "Pas de coût caché et de surprise en fin de mois. Notre grille tarifaire est construite pour être compréhensible.",
+  "pci_projects_project_activate_banner_img_alt": "Discovery Project activation page background image.",
+
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json:17]:
```suggestion
```

File: packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
+  "pci_projects_project_activate_message_with_promotion_success": "Votre projet a bien été activé. Vous bénéficiez de {{ amount }} de crédit offert, valable dans l'univers Public Cloud.",
+  "pci_projects_project_activate_message_without_promotion_success": "Votre projet a bien été activé. Vous pouvez dès à présent démarrer le déploiement de vos ressources Cloud.",
+
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json:20]:
```suggestion
```

File: packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -1,16 +1,22 @@
...
-  "pci_projects_project_pci_banner_promotion_description_2": "Vous serez débité uniquement si vous poursuivez l'utilisation de nos produits au delà de cette somme ou cette date. De quoi démarrer sereinement.",
-  "pci_projects_project_pci_banner_no_promotion_description": "Nos services sont consommables à la demande. Vous pouvez facilement démarrer un service pour le tester quelques heures et l’éteindre dès que vous n’en avez plus besoin.",
-  "pci_projects_project_pci_banner_section_1_title": "Facturation granulaire",
-  "pci_projects_project_pci_banner_section_1_description": "La facturation des ressources se fait en fonction de votre utilisation. Si vous utilisez un produit une heure, vous ne serez facturé que de cette heure là.",
-  "pci_projects_project_pci_banner_section_2_title": "Facturation simple",
-  "pci_projects_project_pci_banner_section_2_description": "Pas de coût caché et de surprise en fin de mois. Notre grille tarifaire est construite pour être compréhensible.",
-  "pci_projects_project_pci_banner_img_alt": "Discovery Project activation page background image."
+  "pci_projects_project_activate_banner_promotion_title": "Profitez de </br> {{ amount }} offerts",
```
[the remark this answers is not in the document]
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json:7]: Yes, the Ui designer insisted to have a break on this part of the phrase.

File: packages/manager/modules/pci/src/projects/creating/service.js [hunk 9]
```diff
@@ -18,8 +18,10 @@ export default class ProjectCreation {
         orderId,
       })
       .$promise.then((detailIds) => {
-        const detailPromises = map(detailIds, (detailId) =>
-          this.OvhApiMeOrder.v6()
+        const detailPromises = map(detailIds, (detailId) => {
```
[the remark this answers is not in the document]
Comment (brofik) [reply packages/manager/modules/pci/src/projects/creating/service.js]: done

File: packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json (hunk 6 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json:17]: done

File: packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json (hunk 7 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/activate/translations/Messages_fr_FR.json]: done

**Review by ghyenne — APPROVED** [review]

**Pull request merged**